lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -952,6 +1001,31 @@ def accessible_operations(user_or_attribute, object_or_attribute, operation_id =
       end
     end
 
+    def all_ancestor_objects(user_or_attribute, operation, root_object, ancestor_objects, options)
+      apply_include_condition!(ancestor_objects, options[:key], options[:includes])
+
+      if !options[:ignore_prohibitions] && prohibition = prohibition_for(operation)
+        prohibited_ancestor_objects = accessible_ancestor_objects(
+          user_or_attribute,
+          prohibition,
+          root_object,
+          options.merge(ignore_prohibitions: true)
+        )
```

Comment:
I might be reading this wrong, but could you add an optional parameter to `accessible_ancestor_objects` that can take the already-loaded ancestor_objects here to prevent requesting them again in https://github.com/mdsol/the_policy_machine/pull/136/files#diff-aae96e1c7c46150cd4f06b06c69b75d3R782?